api/src/test/scala/org/virtuslab/ideprobe/TestScopeParsingTest.scala

```diff
@@ -0,0 +1,49 @@
+package org.virtuslab.ideprobe
+
+import com.typesafe.config.ConfigFactory
+import org.junit.{Assert, Test}
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.virtuslab.ideprobe.protocol.TestScope
+import pureconfig.ConfigSource
+
+@RunWith(classOf[JUnit4])
+final class TestScopeParsingTest {
+
+  private def testParsing(confString: String): TestScope = {
+    val conf = ConfigFactory.parseString(confString)
+    ConfigSource.fromConfig(conf).load[TestScope].getOrElse(
+      error("Parsing error occured")
+    )
+  }
+
+  @Test
+  def parseModule(): Unit = {
+    val scope = testParsing("""{ module: { name: "moduleName" }}""")
+    assert(scope.isInstanceOf[TestScope.Module])
+  }
+
+  @Test
+  def parseDirectory(): Unit = {
+    val scope = testParsing("""{ module: { name: "moduleName" }, directory-name: "/tmp" }""")
+    assert(scope.isInstanceOf[TestScope.Directory])
```

Comment:
Lets make sure values are also correct, it will additionally make it more eye pleasing, without `instanceOf`.
assertEquals(scope, TestScope.Directory(directoryName = "/tmp", ModuleRef("moduleName"))